From patchwork Thu Nov 29 00:00:46 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Jim Mattson <jmattson@google.com>
X-Patchwork-Id: 10703663
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A4A1C13A4
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 29 Nov 2018 00:00:56 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 911472D0EB
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 29 Nov 2018 00:00:56 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 84EA72DB5B; Thu, 29 Nov 2018 00:00:56 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-15.5 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI,
	USER_IN_DEF_DKIM_WL autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 66C362D0EB
	for <patchwork-kvm@patchwork.kernel.org>;
 Thu, 29 Nov 2018 00:00:55 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726740AbeK2LEQ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Thu, 29 Nov 2018 06:04:16 -0500
Received: from mail-io1-f73.google.com ([209.85.166.73]:42454 "EHLO
        mail-io1-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726328AbeK2LEQ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Thu, 29 Nov 2018 06:04:16 -0500
Received: by mail-io1-f73.google.com with SMTP id n22so123607iob.9
        for <kvm@vger.kernel.org>; Wed, 28 Nov 2018 16:00:53 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:message-id:mime-version:subject:from:to:cc;
        bh=LSmA7B0bfi6djFgNShqyWb0llcsk4kYpiZGFi2kFFh8=;
        b=EvAdpy1ynJwjIhR3/sSXCmrQ/v98MftY/WdDOBfhaKijFLRSubXL2ZV51QLEnzq4KK
         VxS9i2FKtL0kTEAbCGY9lhqW4crzRG9aOrfCfKbx08P02lwNByGoqQxDFcmZIJ/Z2sE9
         Ot+X4yAah2/affsjuhR4PM0kw/FqQ4qtd1mM1i4XsFz8gfDTxzTqDqL6oEzqVQr7GMTt
         OWLJ/CkS27wzj5xihI1uggC67MhVFUDCmITlH6HMUmmyyDgleLOUmLUhzlCSPW9XDSsK
         1XiV8Y+B8L4js7JjKgtN7/hn9KMhYDYGw3fpmkdm3r9cmoN9XZI0g4brSLGq6xqXzfRV
         dR8w==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:message-id:mime-version:subject:from:to:cc;
        bh=LSmA7B0bfi6djFgNShqyWb0llcsk4kYpiZGFi2kFFh8=;
        b=fcIc2DL8t7gNOQM8/fQEWrDXUSF1Vz7jTHUqYFZr48sWDnIKjoyiLUXD1ndwZ2dPij
         yfw9SI2WJ7Q3YTstXAI7TU/s8rn/FDxdtxiVdcnhYAF4jxyf9+R2rvCZWWJwp+7GCs6Q
         0MO+cmyBRiumIWjwyEPm9wCeUHvslNwZKAHPjleycUpyjTqa2r+mozqtZXGNQgLIQwau
         U6PPHaSWHKkUUFfcUgRCV5WNCSSni+qtWUOlfKDO2v1/U9ljSGWDxW6g1GYp9WpjZwHK
         /iVA0tV58KZeTOUU2S0SkTwyurzM0RLPQ+NtvPDNAqjrVWbAOI1HTKtrbdFX3phBjc3E
         H9jA==
X-Gm-Message-State: AA+aEWY60I0V3EqOVsw4mMI23zFf8EGgJEj/muubA6Z0bfKVlQIMh+Oa
        E+v/DXAx582uSisK1B4A/E2cKx/DNqZFNUL5RESQX8xCffTQnqa1KHCefXRzLt6dA5QltHNQcft
        VMxYc64O86rYWK0LQJ/D0PYfUiD5wcGCr61OyKbioEH17LpuO3QvFp8qtxa5BMa0=
X-Google-Smtp-Source: 
 AFSGD/WAmG3QJ4UNP/y3aQcREENovdnM1FManQAwdROBQ+iNMfEME9cdC5GlCOacdXcHrY26sdDtPECwxdz6ew==
X-Received: by 2002:a24:4597:: with SMTP id c23mr4028879itd.20.1543449652948;
 Wed, 28 Nov 2018 16:00:52 -0800 (PST)
Date: Wed, 28 Nov 2018 16:00:46 -0800
Message-Id: <20181129000046.93876-1-jmattson@google.com>
Mime-Version: 1.0
X-Mailer: git-send-email 2.20.0.rc1.387.gf8505762e3-goog
Subject: [kvm-unit-tests PATCH] kvm: nVMX: Always VMCLEAR before the first
 VMPTRLD
From: Jim Mattson <jmattson@google.com>
To: kvm@vger.kernel.org
Cc: Jim Mattson <jmattson@google.com>
Content-Type: text/plain; charset="UTF-8"
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Per SDM volume 3, section 24.11.3: Initializing a VMCS, "to avoid the
uncertainties of implementation-specific behavior, software should
execute VMCLEAR on a VMCS region before making the corresponding VMCS
active with VMPTRLD for the first time."

Signed-off-by: Jim Mattson <jmattson@google.com>
Reviewed-by: David Hildenbrand <david@redhat.com>
---
 x86/vmx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/x86/vmx.c b/x86/vmx.c
index ab3c2a4..6ba56bc 100644
--- a/x86/vmx.c
+++ b/x86/vmx.c
@@ -1384,7 +1384,8 @@ static void test_vmptrld(void)
 	       make_vmcs_current(tmp_root) == 1);
 
 	/* Pass VMXON region */
-	make_vmcs_current(vmcs);
+	assert(!vmcs_clear(vmcs));
+	assert(!make_vmcs_current(vmcs));
 	tmp_root = (struct vmcs *)vmxon_region;
 	report("test vmptrld with vmxon region",
 	       make_vmcs_current(tmp_root) == 1);
